app/controllers/api/v1/customers_controller.rb

```diff
@@ -80,9 +80,12 @@ def search_customers
       end
 
       def visible_customers
-        current_api_user.customers.or(
-          Customer.where(enterprise_id: editable_enterprises)
-        )
+        Customer.of(managed_enterprise_ids)
+      end
+
+      def managed_enterprise_ids
+        @managed_enterprise_ids ||= Enterprise.managed_by(current_api_user).
+          select('enterprises.id')
```

Comment:
I can see that this part is copied from the Admin::CustomersController. I agree with David that we should centralise this. It's not about asking for permission of a certain action and therefore I wouldn't use cancancan for this. Instead, it's a collection of records which could go into a scope. What do you think about creating `Customer.managed_by(user)` as a new scope which is then used in both controllers?

Reply:
I agree. 
I implemented the new scope.